### PR TITLE
macOS internationalization of xibs

### PR DESCRIPTION
**Pull request by sweetppro**
initial update of project to use Base.lproj localization
#3405

### CODE REVIEW

**Comment by nevack** [comment]
@sweetppro You need to update CMakeLists :)

**Comment by sweetppro** [comment]
@nevack oops, on it

**Comment by sweetppro** [comment]
hmm, im a little lost on how to update the CMakelists file
any pointers @nevack

**Comment by nevack** [comment]
> hmm, im a little lost on how to update the CMakelists file
> any pointers @nevack

I can try to help you there.
Do you mind me pushing new comments into this branch?

**Comment by sweetppro** [comment]
> I can try to help you there. Do you mind me pushing new comments into this branch?

no of course not, much appreciated

**Comment by nevack** [comment]
@ckerr `alpine-musl` job is failing with
```
In file included from /__w/transmission/transmission/src/cli/cli.cc:20:
[251](https://github.com/transmission/transmission/runs/7187597894?check_suite_focus=true#step:9:252)
ninja: subcommands failed
[252](https://github.com/transmission/transmission/runs/7187597894?check_suite_focus=true#step:9:253)
/__w/transmission/transmission/src/libtransmission/utils.h:37:10: fatal error: libintl.h: No such file or directory
[253](https://github.com/transmission/transmission/runs/7187597894?check_suite_focus=true#step:9:254)
   37 | #include <libintl.h>
[254](https://github.com/transmission/transmission/runs/7187597894?check_suite_focus=true#step:9:255)
      |          ^~~~~~~~~~~
[255](https://github.com/transmission/transmission/runs/7187597894?check_suite_focus=true#step:9:256)
compilation terminated.
[256](https://github.com/transmission/transmission/runs/7187597894?check_suite_focus=true#step:9:257)
Error: Process completed with exit code 1.
```

**Comment by sweetppro** [comment]
If these 2 PRs look ok, ill get started on converting the remains xibs
The first is the MainMenu.xib conversion: #3411

**Comment by sweetppro** [comment]
ok, so Ive converted all the xibs which were already in auto layout.
I need to step out now,
but i'll start converting the remaining xibs to auto layout tonight

**Comment by sweetppro** [comment]
just the Preferences window left,
this one could take some time...

**Comment by sweetppro** [comment]
Closes #3039

**Comment by sweetppro** [comment]
ok
so all the required xibs have been converted to auto layout.
**Id appreciate if people could now test to see if I've missed something.**

There are a couple of issues in the **Preferences window** I've noticed which do need attention _(any help appreciated here)_:
* The max width of all Tabs needs to be computed before opening the Preferenecs  window _(This is an issue only in non-English runtimes)_. Otherwise, on first opening of each tab - the width will be calculated dynamically - creating an undesirable width animation. _(only occurs until we have found the max width required)_
* The change in height animation when changing tabs can be clunky. I have a feeling this could be a similar issue with the Height needing to be calculated at runtime. _(This occurs in all languages)_

**Comment by Oleg-Chashko** [comment]
@sweetppro 
If you give me a ready build, I'll check it out. When I get there.

**Comment by sweetppro** [comment]
@Oleg-Chashko ive just stepped out to workout. I’ll upload a build after lunch for you

**Comment by Oleg-Chashko** [comment]
@sweetppro 
macOS - Button artifacts appear when "Show Toolbar" is turned on #1903

**Comment by Oleg-Chashko** [comment]
@sweetppro 
Screen Shot 2022-07-08 at 14 13 09

**Comment by Oleg-Chashko** [comment]
@sweetppro 
In order to reproduce this error, a certain sequence of actions is required!

https://user-images.githubusercontent.com/62497891/177991574-a4db5134-fb89-4595-86a5-51b0672e9ec8.mp4

**Comment by sweetppro** [comment]
> @sweetppro
> macOS - Button artifacts appear when "Show Toolbar" is turned on #1903

this seems to only be the case when Autosize is disabled. let me look over the previous PR

**Comment by Oleg-Chashko** [comment]
@sweetppro

First I will publish all the errors, and then I will answer the questions.

**Comment by Oleg-Chashko** [comment]
@sweetppro 

Screen Shot 2022-07-08 at 15 55 16

Screen Shot 2022-07-08 at 15 54 58

**Comment by Oleg-Chashko** [comment]
@sweetppro 

Back to the way it was about three weeks ago. 
On 13 inches, it took up more than half of the display. 

[macOS] Reduce the width of the buttons on the toolbar #3375

Screen Shot 2022-07-08 at 16 17 55

**Comment by sweetppro** [comment]
> Back to the way it was about three weeks ago. On 13 inches, it took up more than half of the display.
I reverted this, as there were complaints the Toolbar was not very Mac like

**Comment by Oleg-Chashko** [comment]
@sweetppro 

https://user-images.githubusercontent.com/62497891/178014593-a2f49315-8af0-499d-9dd9-608eba2f785c.mp4

**Comment by Oleg-Chashko** [comment]
@sweetppro 

The width is limited and can not expand to fit the words.

Screen Shot 2022-07-08 at 16 54 27

**Comment by Oleg-Chashko** [comment]
@sweetppro 

Screen Shot 2022-07-08 at 17 19 13

**Comment by Oleg-Chashko** [comment]
@sweetppro 
Russian localization:
Screen Shot 2022-07-08 at 17 30 32

Screen Shot 2022-07-08 at 17 31 52

Screen Shot 2022-07-08 at 17 34 15

Screen Shot 2022-07-08 at 17 37 23

Screen Shot 2022-07-08 at 18 22 12

Screen Shot 2022-07-08 at 17 41 13

**Comment by sweetppro** [comment]
A lot of those issues I’ve fixed already (maybe all of them)

**Comment by Oleg-Chashko** [comment]
@sweetppro 

Make the distance between the checkboxes the same. In all tabs.

Screen Shot 2022-07-08 at 18 37 01

**Comment by sweetppro** [comment]
Im happy with this now.
The major bugs are worked out, I think it's ready for more eyes

@Oleg-Chashko here you go, the latest build:
https://send.tresorit.com/a#UUzIuy2pYiqFHQA6u59wIg

**Comment by Oleg-Chashko** [comment]
Screen Shot 2022-07-08 at 20 00 53

**Comment by sweetppro** [comment]
thats a weird ui glitch.... looking into it now

**Comment by Oleg-Chashko** [comment]
Screen Shot 2022-07-08 at 20 24 36

**Comment by Oleg-Chashko** [comment]
Screen Shot 2022-07-08 at 20 28 58

**Comment by Oleg-Chashko** [comment]
Screen Shot 2022-07-08 at 20 30 43

**Comment by Oleg-Chashko** [comment]
Screen Shot 2022-07-08 at 20 28 58

**Comment by Oleg-Chashko** [comment]
When you fix everything I posted, only then will you send me the download link.

**Comment by Oleg-Chashko** [comment]
Screen Shot 2022-07-08 at 20 46 10
Screen Shot 2022-07-08 at 20 44 17

**Comment by sweetppro** [comment]
here you go,
I think this fixes all those. except for the empty autoresize window size.
I couldn't recreate that
https://send.tresorit.com/a#jSlra1fHS8XylEnBwZjNIg

Im done for today, If theres any remaining issues, they can wait till tomorrow
:)

@Oleg-Chashko  thx for the thorough testing and feedback

**Comment by Oleg-Chashko** [comment]
**Not fixed yet!**

---

1. macOS - Button artifacts appear when "Show Toolbar" is turned on https://github.com/transmission/transmission/issues/1903

---

2. In order to reproduce this error, a certain sequence of actions is required!

https://user-images.githubusercontent.com/62497891/178063966-eb1f598c-f354-46b6-9e13-a237150330b1.mp4

---

3. Put the turtle down not much.
    (to) lift upwards not much.



---

4. Remove the empty space.

**Comment by sweetppro** [comment]
1. fixed in #3419
3. both are vertically centre aligned via Xcode

**Comment by Oleg-Chashko** [comment]
I apologize for the off-topic. 
The difficulty of the work is that I only know Ukrainian (100), Russian (100), German 50/50, English only read text.

**Comment by Oleg-Chashko** [comment]
@sweetppro 

A request from me, if it is possible to consider this issue.

reduce wasted space in macOS client's Torrent Inspector UI #3223

**Comment by sweetppro** [comment]
Ive fixed most issues:
4. while apparent in this PR, im not sure how as I havnt touched the code or xib related to that particular Info tab.... im looking into it though
13. is simply not feasible, with static window widths

**Comment by Oleg-Chashko** [comment]
> Ive fixed most issues:
> 13. is simply not feasible, with static window widths

I don't have number 13. Could you explain what you meant by that?

**Comment by sweetppro** [comment]
> > Ive fixed most issues:
> > 13. is simply not feasible, with static window widths
> 
> I don't have number 13. Could you explain what you meant by that?

that should have been 12...

File: macosx/InfoOptionsViewController.mm [hunk 1]
```diff
@@ -79,13 +79,54 @@ - (void)awakeFromNib
...
 }
 
 - (void)dealloc
 {
     [NSNotificationCenter.defaultCenter removeObserver:self];
 }
 
+- (void)updateWindowLayout
+{
+    CGFloat inset = 12.0, spacing = 8.0, difference = 0;
```
[the remark this answers is not in the document]
Comment (sweetppro) [reply macosx/InfoOptionsViewController.mm:93]: These values are defined in the xib.

The **inset** value was already defined, and the **spacing** value was the default value added to the NSStackView by Xcode.

[the remark this answers is not in the document]
Comment (sweetppro) [reply macosx/InfoOptionsViewController.mm]: Ive refactored this code

**Comment by sweetppro** [comment]
English UI Demo:
https://imgur.com/MUvFTFH

Geman UI Demo:
https://imgur.com/Ophs3jI

**Comment by sweetppro** [comment]
@Oleg-Chashko here's the final build:
https://send.tresorit.com/a#mnVwFfq_aI4PENEqVqoy0A

**Review by nevack — APPROVED** [review]
LGTM
Great job!

**Comment by sweetppro** [comment]
@nevack Thx!
It took a bit more work than I originally anticipated...
C'est la Vie

**Comment by Oleg-Chashko** [comment]
@sweetppro 

Checked what was fixed, I deleted. What was not fixed remained in my report. 

---

Reduce wasted space in macOS client's Torrent Inspector UI #3223
Better to remove from this РR. We still have to deal with it. It is a little crooked.

**Comment by Oleg-Chashko** [comment]
> @Oleg-Chashko I fixed all your issues.

It seems to you. Screenshot of your desktop, it proves you wrong. 

Screen Shot 2022-07-14 at 17 37 14

**Comment by sweetppro** [comment]
@Oleg-Chashko try the newest build I uploaded, in the comment directly before yours that screen shot is from the video from the older build

**Comment by sweetppro** [comment]
> I hope you don't make these changes to this PR.

@Oleg-Chashko Im not sure I understand...
you asked for these changes in #3223 

what exactly is wrong with the implementation?
This auto layout PR is a complete overhaul of the UI,
why not include it here?

**Comment by Oleg-Chashko** [comment]
> > I hope you don't make these changes to this PR.
> 
> @Oleg-Chashko Im not sure I understand... you asked for these changes in #3223
> 
> what exactly is wrong with the implementation? This auto layout PR is a complete overhaul of the UI, why not include it here?


I've written to you about this before.
https://github.com/transmission/transmission/pull/3409#issuecomment-1184380338

Yes, there are problems with different localization as well as the rest of the execution. Because of this, it is better to do a different PR. I think it's up to you to decide what's best for you.

**Comment by sweetppro** [comment]
As far as I am aware all issues have now been resolved in this PR

> Remove the empty space.
> 
> 

This is fixed in the latest commit, as well a potential window sizing issue in the Info panel when switching to either the **Options** or **Activity** tabs

**Comment by ckerr** [comment]
#

**Comment by sweetppro** [comment]
Closes #3405

**Comment by ckerr** [comment]
@sweetppro  it sounds like this is ready to be merged but I thought I'd ask first to be sure?

**Comment by sweetppro** [comment]
@ckerr as far as I am aware it is ready, all issues relayed to me were fixed

**Pull request merged**